identity/src/main/java/com/stripe/android/identity/IdentityActivity.kt

```diff
@@ -71,14 +80,47 @@ internal class IdentityActivity :
 
     private var launchedFallbackUrl: Boolean = false
 
+    lateinit var subcomponent: IdentityActivitySubcomponent
+
+    @Inject
+    lateinit var subComponentBuilderProvider: Provider<IdentityActivitySubcomponent.Builder>
+
+    @Inject
+    @UIContext
+    lateinit var uiContext: CoroutineContext
+
+    @Inject
+    @IOContext
+    lateinit var workContext: CoroutineContext
+
+    override fun fallbackInitialize(arg: Context) {
+        DaggerIdentityActivityComponent.builder()
+            .context(arg)
+            .build().inject(this)
+    }
+
     override fun onSaveInstanceState(outState: Bundle) {
         super.onSaveInstanceState(outState)
         outState.putBoolean(KEY_LAUNCHED_FALLBACK_URL, launchedFallbackUrl)
         outState.putBoolean(KEY_PRESENTED, true)
     }
 
     override fun onCreate(savedInstanceState: Bundle?) {
-        supportFragmentManager.fragmentFactory = identityViewModel.identityFragmentFactory
+        injectWithFallback(
+            starterArgs.injectorKey,
+            this
+        )
+        subcomponent = subComponentBuilderProvider.get()
+            .args(starterArgs)
+            .cameraPermissionEnsureable(this)
+            .appSettingsOpenable(this)
+            .verificationFlowFinishable(this)
+            .identityViewModelFactory(viewModelFactory as IdentityViewModel.IdentityViewModelFactory)
+            .fallbackUrlLauncher(this)
+            .build()
```

Comment:
Does this mean we'll be recreating the subcomponent on configuration changes (i.e: screen rotation)?

Reply:
That's true, although we have locked the activity to portrait for our SDK due to the Camera UI.
I think the `subcomponent` has to be recreated in `onCreate` inorder for `injectWithFallback` to work - when fallback happens, there will be no "bigger" scope to instantiate the subcomponent, we'll need to create a `DaggerIdentityActivityFallbackComponent` from scratch. 

We also cannot save the subcomponent in the ViewModel due to the reason stated in the PR description :(